plugins/modules/sp_client_policy.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# (c) 2024,Tom page <tpage@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'], 'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: sp_client_policy
+author: "Subhajit Patra (@subhpa01)"
+short_description: Configure backup schedule policies for IBM Storage Protect clients
+description:
+    - This module defines backup schedules and assigns client nodes in IBM Storage Protect.
+options:
+    schedule_name:
+        description: Name of the backup schedule.
+        required: true
+        type: str
+    schedule_type:
+        description: Type of the schedule (clientaction/admin).
+        required: true
+        type: str
+        choices: ["clientaction", "admin"]
+    start_time:
+        description: Schedule start time (HH:MM:SS format).
+        required: true
+        type: str
+    period:
+        description: Period between backups.
+        required: true
+        type: int
+    period_unit:
+        description: Unit of the period (hours/days).
+        required: true
+        type: str
+        choices: ["hours", "days"]
+    client_nodes:
+        description: List of client nodes to associate with the schedule.
+        required: true
+        type: list
+        elements: str
+extends_documentation_fragment: ibm.storage_protect.auth
+...
+'''
+
+
+EXAMPLES = '''
+- name: Configure backup schedule for client nodes
+  ibm.storage_protect.sp_client_policy:
+    schedule_name: "{{ DailyBackup }}"
+    schedule_type: "{{ clientaction }}"
+    start_time: "22:00:00"
+    period: 1
+    period_unit: "days"
+    client_nodes:
+      - "node1"
+      - "node2"
+...
+'''
+
+from ..module_utils.dsmadmc_adapter import DsmadmcAdapter
+
+
+def main():
+    argument_spec = dict(
+        schedule_name=dict(type="str", required=True),
+        schedule_type=dict(type="str", required=True, choices=["clientaction", "admin"]),
+        start_time=dict(type="str", required=True),
+        period=dict(type="int", required=True),
+        period_unit=dict(type="str", required=True, choices=["hours", "days"]),
+        client_nodes=dict(type="list", elements="str", required=True)
```

Comment:
I don't think any of these should be `required=True`. They all have defaults on the dsmadmc command